src/js/services/StorageService.js

```diff
@@ -0,0 +1,17 @@
+// This is the service to interact with the Chrome storage
+// We can switch this service to IndexedDBService if we want to use IndexedDB instead of Chrome storage
+export class StorageService {
+  async get(key, defaultValue = null) {
```

Comment:
try catch {} especially important because if the database operations like `chrome.storage.sync.get` , etc. fail, it'll be hard to map these errors all the way down to the DB layer.

Reply:
Agreed! This is great will update accordingly